src/gui/src/components/navigation/header.tsx

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from 'react'
+import { useGraphStore } from '@/state/index.js'
+
+const Header = () => {
+  const [routeName, setRouteName] = useState<string>('')
+  const route = useGraphStore(state => state.activeRoute)
+
+  useEffect(() => {
+    /**
+     * Set a clean route on the state for display
+     */
+    if (location.pathname === '/') {
+      setRouteName('Dashboard')
+    } else if (route) {
+      const parts = route.split('/')
+      const cleanedRoute =
+        (parts[1]?.charAt(0).toUpperCase() || '') +
+        (parts[1]?.slice(1) || '')
+      setRouteName(cleanedRoute)
```

Comment:
[optional] I think having a simpler map here will be more flexible so that we can set page titles independent from the route path, so that in the future we could have routes such as `/explore/manage` and have the title be something arbitrary like `Manage your dependencies!`, e.g:

```suggestion
      const defaultRouteName = 'Dashboard'
      const routeNames = new Map<string, string>([
        ['/error', 'Error'],
        ['/explore', 'Explore'],
      ])
      const routeName = routeNames.get(route) || defaultRouteName
      setRouteName(routeName)
```